### PR TITLE
YLP-1805 Push error to the backend

### PR DESCRIPTION
**Pull request by TimCanu**

### CODE REVIEW

File: packages/yourloops/app/error.tsx [hunk 1]
```diff
@@ -75,52 +98,43 @@ function OnError(props: OnErrorProps): JSX.Element {
...
     <Dialog
-      id="dialog-app-crash"
-      open
+      open={true}
       fullScreen={fullScreen}
       fullWidth
       maxWidth="sm"
-    >
-      <DialogTitle id="dialog-app-crash-title">{t('app-crash-title')}</DialogTitle>
+     >
```
Comment (celia-rouis) [inline packages/yourloops/app/error.tsx:107]: Why is it not aligned with `<Dialog` anymore? Was it on purpose?

File: packages/yourloops/app/error.tsx [hunk 2]
```diff
@@ -75,52 +98,43 @@ function OnError(props: OnErrorProps): JSX.Element {
...
+            />
+          </>
+        ) : (
+          <Button
+            variant="text"
+            color="primary"
+            onClick={handleShowMore}
+          >
+            {t('app-crash-button-more-info')}
+          </Button>)
```
Comment (celia-rouis) [inline packages/yourloops/app/error.tsx:136]: New line for the parenthesis? (symmetry with the other one)

File: packages/yourloops/app/error.tsx [hunk 3]
```diff
@@ -48,17 +52,36 @@ interface OnErrorProps {
...
     try {
       metrics.send('error', 'app-crash', errorMessage)
+      ErrorApi.sendError({
+        browserName,
+        browserVersion,
+        date: moment(new Date()).format('DD/MM/YYYY'),
+        err: errorMessage,
+        errorId,
+        path: location.pathname
+      }).catch(error => console.error(error))
```
Comment (celia-rouis) [inline packages/yourloops/app/error.tsx:84]: I don't understand why there is a need for both `ErrorApi.sendError().catch()` and `try {} catch {}`? Besides, they seem to be doing the same thing

File: packages/yourloops/app/error.tsx [hunk 4]
```diff
@@ -48,17 +52,36 @@ interface OnErrorProps {
...
   const { t } = useTranslation('yourloops')
   const theme = useTheme()
+  const location = useLocation()
   const [showMore, setShowMore] = React.useState(false)
   const fullScreen = useMediaQuery(theme.breakpoints.down('md'))
-
+  const errorId = uuidv4()
+  const style = classes()
   const errorMessage = props.error?.message ?? 'n/a'
+  const error = props.error ? `Error: ${errorMessage}\nStack: ${props.error.stack}` : 'N/A'
```
Comment (celia-rouis) [inline packages/yourloops/app/error.tsx:71]: Why is the fallback error an uppercase `'N/A'` when the fallback error message is lowercase `'n/a'`? Maybe the same value in both cases would be better?

File: packages/yourloops/app/error.tsx (hunk 3 above)
Comment (TimCanu) [reply packages/yourloops/app/error.tsx:84]: The try catch cannot catch the error coming from `ErrorApi.sendError(...)`.
And the try catch is used to get the error for the `metrics.send(...)`
But I think that I can improve this as our http service seems to log the error too. :slightly_smiling_face:

**Review by ydrisR — APPROVED** [review]
LGTM

**Pull request merged**